src/NUnitEngine/nunit.engine.tests/Services/TestFilteringTests.cs

```diff
@@ -34,6 +35,7 @@ namespace NUnit.Engine.Services.Tests
     public class TestFilteringTests
     {
         private const string MOCK_ASSEMBLY = "mock-assembly.dll";
+        private const string NUNIT_FRAMEWORK_NAME = "nunit.framework, Version=3.9.0.0, Culture=neutral, PublicKeyToken=2638cd05610744eb";
```

Comment:
Same

Reply:
> Also, HintPath doesn't meet my definition of 'hardcoding'

It's not the hint path that's hardcoded, it's this part: `<Reference Include="nunit.framework, Version=3.9.0.0, Culture=neutral, PublicKeyToken=2638cd05610744eb, processorArchitecture=MSIL">` combined with the fact that there is no SpecificVersion false tag (for good reason).

Pulling the reference out of the assembly (using Mono.Cecil, I assume you mean?) sounds fine to me, I can update the PR.